test/unit/rules/sort-invocations-test.js

```diff
@@ -0,0 +1,2027 @@
+          )
+          data-test-button=""
+        }}
+          Submit form
+        {{/ui/button}}
+      `,
+      fixedTemplate: `
+        <Ui::Button
+          @isDisabled={{not this.enableSubmit}} @label="Submit form" @type="submit" class={{local
+            this.styles
+            "button"
+            (unless this.enableSubmit "disabled")
+          }} data-cucumber-button="Submit form" data-test-button ...attributes {{autofocus}} {{on "click" @onSubmit}}
+        />
+
+        <Ui::Button
+          @isDisabled={{not this.enableSubmit}} @type="submit" class={{local
+            this.styles
+            "button"
+            (unless this.enableSubmit "disabled")
+          }} data-cucumber-button="Submit form" data-test-button={{""}} ...attributes {{autofocus}} {{on "click" @onSubmit}}
+        >
+          Submit form
+        </Ui::Button>
+
+        {{ui/button
+          class=(local
+            this.styles
+            "button"
+            (unless this.enableSubmit "disabled")
+          )
+          data-cucumber-button="Submit form"
+          data-test-button=""
+          isDisabled=(not this.enableSubmit)
+          label="Submit form"
+          onclick=onSubmit
+          type="submit"
+        }}
+
+        {{#ui/button
+          class=(local
+            this.styles
+            "button"
+            (unless this.enableSubmit "disabled")
+          )
+          data-cucumber-button="Submit form"
+          data-test-button=""
+          isDisabled=(not this.enableSubmit)
+          onclick=onSubmit
+          type="submit"
+        }}
+          Submit form
+        {{/ui/button}}
+      `,
+      verifyResults(results) {
+        expect(results).toMatchInlineSnapshot(`
+          [
+            {
+              "column": 8,
+              "endColumn": 10,
+              "endLine": 16,
+              "filePath": "layout.hbs",
+              "isFixable": true,
+              "line": 2,
+              "message": "\`data-cucumber-button\` must appear later",
+              "rule": "sort-invocations",
+              "severity": 2,
+              "source": "<Ui::Button
+                    data-cucumber-button="Submit form"
+                    {{on "click" @onSubmit}}
+                    @type="submit"
+                    @isDisabled={{not this.enableSubmit}}
+                    class={{local
+                      this.styles
+                      "button"
+                      (unless this.enableSubmit "disabled")
+                    }}
+                    ...attributes
+                    data-test-button
+                    @label="Submit form"
+                    {{autofocus}}
+                  />",
+            },
+            {
+              "column": 8,
+              "endColumn": 10,
+              "endLine": 16,
+              "filePath": "layout.hbs",
+              "isFixable": true,
+              "line": 2,
+              "message": "\`on\` must appear later",
```

Comment:
`must appear later` is not a valuable message, we should point after that it should appear (in my understanding)
Like: 
`"on" modifier should appear after data-test-name attribute` or
`{{on}} modifier should appear before "...attributes"` or
`"id" attribute should appear after "ab" attribute" or
`@name argument should appear after "id" attribute`

Reply:
I don't think we can easily implement such precise before and after's. This would require a more drastic change to the code that I wrote. The existing `sort-invocations` also [doesn't specify where exactly something should go](https://github.com/ember-template-lint/ember-template-lint/blob/v7.6.0-ember-template-lint/test/unit/rules/attribute-order-test.js#L106-L118).

`sort-invocations` is really meant to be followed using autofix, so I think the log message can remain brief as is. I'll will surround the modifier names with `{{}}` to more clearly indicate things that are modifiers.

Reply:
Nvm. We can easily write the after. :)